Node JS/session fswd 01/UI/food-adda/src/components/Signup.jsx

```diff
@@ -0,0 +1,149 @@
+
+    // Create a FormData object
+    const formData = new FormData();
+    formData.append("email", email);
+    formData.append("password", pswd);
+    formData.append("userType", userType);
+
+    const data = Object.fromEntries(formData);
+    console.log(data);
+    // Send the data to the backend using axios
+    // axios.post("http://localhost:5000/api/users/signup", data)
+    //     .then(res => {
+    //         console.log(res);
+    //     })
+    //     .catch(err => {
+    //         console.log(err);
+    //     })
+
+    // Close the modal and toast a success message
+    toast.success("Signed up successfully!", {
+      position: "top-center",
+      autoClose: 2000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      progress: undefined,
+    });
+    props.onHide();
+  };
+
+  return (
+    <div>
+      <Modal
+        {...props}
+        size="lg"
+        aria-labelledby="contained-modal-title-vcenter"
+        centered
+      >
+        <Modal.Body>
+          <AiOutlineCloseCircle
+            size={24}
+            style={{
+              position: "absolute",
+              top: "10px",
+              right: "10px",
+              cursor: "pointer",
+            }}
+            onClick={props.onHide}
+          />
+          <h4>Sign Up</h4>
+          {/* Sign up form goes here */}
+          <div className="signup-container d-flex flex-column align-items-center">
+            <img
+              src="https://images.unsplash.com/photo-1599305445671-ac291c95aaa9?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8bG9nb3xlbnwwfHwwfHw%3D&auto=format&fit=crop&w=800&q=60"
+              className="w-75 rounded"
+              style={{ maxWidth: "300px" }}
+              alt=""
+            />
+            <Form className="mt-4 w-100" onSubmit={handleSubmit}>
+              {/* A react bootstrap switch to choose between Chef or Customer */}
+              <div className="d-flex justify-content-between mb-3">
+                <Form.Label className="h5">Register as:</Form.Label>
+                <div>
+                  <Form.Check
+                    inline
+                    label="Chef"
+                    name="group1"
+                    value={"chef"}
+                    type="radio"
+                    id="radio-1"
+                    required
+                  />
+                  <Form.Check
+                    inline
+                    label="Customer"
+                    name="group1"
+                    value={"customer"}
+                    type="radio"
+                    id="radio-2"
+                  />
+                </div>
+              </div>
+              <Form.Group className="mb-3" controlId="registerMail">
+                <Form.Control type="email" placeholder="Enter email" />
+              </Form.Group>
+
+              {/* Password not matching alert */}
+              {pswd !== confirmPswd && confirmPswd !== "" ? (
```

Comment:
You may try to control the floating/dance of the container. when this is visible/hidden

Reply:
Or, you can enable the submit button and show the toast message whenever there is any issue

Reply:
avoid validation onKeyUp,